src/cocotb/triggers.py

```diff
@@ -783,19 +805,19 @@ def _prime(self, callback: Callable[[Trigger], None]) -> None:
     def __repr__(self) -> str:
         return f"Join({self._task!s})"
```

Comment:
```suggestion
        return f"TaskComplete({self._task!s})"
```

Reply:
Will also need to update `test_task_repr`.